ext/tokenizer/tokenizer_data.c

```diff
@@ -53,6 +53,8 @@ void tokenizer_register_constants(INIT_FUNC_ARGS) {
 	REGISTER_LONG_CONSTANT("T_COALESCE", T_COALESCE, CONST_CS | CONST_PERSISTENT);
 	REGISTER_LONG_CONSTANT("T_BOOLEAN_OR", T_BOOLEAN_OR, CONST_CS | CONST_PERSISTENT);
 	REGISTER_LONG_CONSTANT("T_BOOLEAN_AND", T_BOOLEAN_AND, CONST_CS | CONST_PERSISTENT);
+	REGISTER_LONG_CONSTANT("T_AMPERSAND_NOT_FOLLOWED_BY_VAR_OR_VARARG", T_AMPERSAND_NOT_FOLLOWED_BY_VAR_OR_VARARG, CONST_CS | CONST_PERSISTENT);
```

Comment:
Some php projects such as http://github.com/nikic/php-parser and https://github.com/microsoft/tolerant-php-parser depend on these constants. So do phpcs, etc.

While it's possible to work around and this does make the fact that the constant was removed (and the application or library needs to be updated) obvious, I think it'd minimize the BC break if T_AMPERSAND was an alias for the constant T_AMPERSAND_NOT_FOLLOWED_BY_VAR_OR_VARARG in php 8.1, especially if a project uses T_AMPERSAND in a large array declaration but never does anything with it.

(This doesn't really affect me personally, it's possible to work around by declaring new constants conditionally using the old constants to avoid conflicts, e.g. https://github.com/microsoft/tolerant-php-parser/blob/main/src/PhpTokenizer.php (works but could be cleaner))

@nikic thoughts on whether removing the constant is a bc break? Would using an alias be appropriate, or making T_AMPERSAND a placeholder value distinct from both of these new token types, and stopping returning T_AMPERSAND from token_get_all entirely?

Reply:
Oh, I didn't realize this, should update the RFC to include this detail, but I think adding an alias is sensible. In a perfect world we wouldn't need these two tokens

Reply:
I've aliased T_AMPERSAND to be `T_AMPERSAND_FOLLOWED_BY_VAR_OR_VARARG` I would imagine this shouldn't break any existing tooling.